File: Row with max 1s.cpp
class Solution {
  public:
    int rowWithMax1s(vector<vector<int> > &arr) {
        // code here
        
        int n=arr.size();
        int m=arr[0].size();
        int mx=-1 , ans=-1;
        
        for(int i=0; i<n;i++){
            int flag=0;
            for(int j=0; j<m ; j++){
                if(arr[i][j]==1 && mx!=m-j){
                    mx=max(mx,m-j);
                    flag=1;
                }
                if(mx==m-j && flag==1){
                    ans=i;
                    break;
                }
            }
        }
        
        return ans;
    }
};

File: Row with max 1s.java
class Solution {
    public int rowWithMax1s(int arr[][]) {
        int n = arr.length;
        if (n == 0) return -1;
        int m = arr[0].length;
        
        int idx = -1;
        int j = m - 1; 

        for (int i = 0; i < n; i++) {
            while (j >= 0 && arr[i][j] == 1) {
                j--; 
                idx = i;
            }
        }

        return idx;
    }
}
